test/postControllerTest.js

```diff
@@ -0,0 +1,148 @@
+const express = require('express');
+const request = require('supertest');
+const postController = require('../controllers/postController');
+const sinon = require('sinon');
+const postService = require('../services/postService');
+const assert = require('assert');
+
+const app = express();
+
+app.set('view engine', 'ejs');
+app.use(express.static('../public'));
+
+app.use('', postController);
+
+describe('postController', () => {
+  beforeEach(() => {
+    sinon.stub(postService, 'getAllPosts').resolves('expected');
+    sinon.stub(postService, 'getOnePost').resolves('expected');
+    sinon.stub(postService, 'createNewPost').resolves('expected');
+  });
+
+  afterEach(() => {
+    postService.getAllPosts.restore();
+    postService.getOnePost.restore();
+    postService.createNewPost.restore();
+  });
+
+  describe('GET /', () => {
+    it('should call method "postService.getAllPosts" with no arguments', () => (
+      request(app)
+        .get('/')
+        .then(() => {
+          sinon.assert.calledWith(postService.getAllPosts);
+        })
+    ));
+  });
+  describe('GET /', () => {
+    it('should return correct "posts" in promise', () => (
+      request(app)
+        .get('/')
+        .then(() => {
+          postService.getAllPosts().then((posts) => {
+            assert.equal(posts, 'expected');
+          });
+        })
+    ));
+  });
+  describe('GET /posts', () => {
+    it('should call method "postService.getAllPosts" with no arguments', () => (
+      request(app)
+        .get('/posts')
+        .then(() => {
+          sinon.assert.calledWith(postService.getAllPosts);
+        })
+    ));
+  });
+  describe('GET /posts', () => {
+    it('should return correct "posts" in promise', () => (
+      request(app)
+        .get('/posts')
+        .then(() => {
+          postService.getAllPosts().then((posts) => {
+            assert.equal(posts, 'expected');
+          });
+        })
+    ));
+  });
+  describe('GET /posts/:post_id', () => {
+    it('should call method "postService.getOnePost" with correct arguments', () => (
+      request(app)
+        .get('/posts/post_id')
+        .then(() => {
+          sinon.assert.calledWith(postService.getOnePost, 'post_id');
+        })
+    ));
+  });
+  describe('GET /posts/:post_id', () => {
+    it('should return correct "post" in promise', () => (
+      request(app)
+        .get('/post/post_id')
+        .then(() => {
+          postService.getOnePost().then((post) => {
+            assert.equal(post, 'expected');
+          });
+        })
+    ));
+  });
+  describe('POST /post_new', () => {
+    it('should call method "postService.createNewPost" with correct arguments', () => (
+      request(app)
+        .post('/post_new')
+        .type('form')
+        .send({ title: 'title', author: 'author', body: 'body' })
+        .then(() => {
+          sinon.assert.calledWith(postService.createNewPost, 'title', 'author', 'body');
+        })
+    ));
+  });
+
+  // Nav Tests
```

Comment:
I'm not too sure what to think about these tests. I do think that testing whether your views are including the nav is a good idea, but it seems a bit weird. 

I think this isn't worth debating over since you are moving to react anyways